q2_types/feature_data/_transformer.py

```diff
@@ -402,3 +407,119 @@ def _224(data: pd.DataFrame) -> DifferentialFormat:
     ff = DifferentialFormat()
     qiime2.Metadata(data).save(str(ff))
     return ff
+
+
+# Protein FASTA transformers
+
+class ProteinIterator(collections.abc.Iterable):
+    def __init__(self, generator):
+        self.generator = generator
+
+    def __iter__(self):
+        yield from self.generator
+
+
+class AlignedProteinIterator(ProteinIterator):
+    pass
+
+
+def _read_protein_fasta(path):
+    return skbio.read(path, format='fasta', constructor=skbio.Protein)
```

Comment:
I think we should drop this (or drop the similarly named method on `AlignedProteinFASTAFormat`. My vote is here, because you can just call the format's method everywhere you were calling this utility.